clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -404,7 +494,16 @@ private void updateGroupSubscription(Set<String> topics) {
 
         log.debug("Performing assignment using strategy {} with subscriptions {}", assignor.name(), subscriptions);
 
-        Map<String, Assignment> assignment = assignor.assign(metadata.fetch(), subscriptions);
+        Map<String, Assignment> assignments = assignor.assign(metadata.fetch(), subscriptions);
+
+        switch (protocol) {
+            case EAGER:
+                break;
+
+            case COOPERATIVE:
+                adjustAssignment(ownedPartitions, assignments);
```

Comment:
A brief description of the purpose of this adjustment would be helpful.

Reply:
Ack.